tiled/client/node.py

```diff
@@ -543,6 +548,111 @@ def _ipython_key_completions_(self):
             # Do not print messy traceback from thread. Just fail silently.
             return []
 
+    def write_array(self, array, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: write an array
+
+        This is subject to change or removal without notice
+
+        """
+
+        from ..structures.array import ArrayMacroStructure, ArrayStructure, BuiltinDtype
+
+        metadata = metadata or {}
+        specs = specs or []
+        mimetype = "application/octet-stream"
+
+        structure = ArrayStructure(
+            macro=ArrayMacroStructure(
+                shape=array.shape,
+                # just one chunk for now...
+                chunks=tuple((size,) for size in array.shape),
+            ),
+            micro=BuiltinDtype.from_numpy_dtype(array.dtype),
+        )
+        data = {
+            "metadata": metadata,
+            "structure": asdict(structure),
+            "structure_family": StructureFamily.array,
+            "specs": specs,
+            "mimetype": mimetype,
+        }
+
+        full_path_meta = (
+            "/node/metadata"
+            + "".join(f"/{part}" for part in self.context.path_parts)
+            + "".join(f"/{part}" for part in (self._path or [""]))
+        )
+        document = self.context.post_json(full_path_meta, data)
+        uid = document["uid"]
+
+        full_path_data = (
+            "/array/full"
+            + "".join(f"/{part}" for part in self.context.path_parts)
+            + "".join(f"/{part}" for part in self._path)
+            + "/"
+            + uid
+        )
+        self.context.put_content(full_path_data, content=array.tobytes())
```

Comment:
Should we send `npy` instead? The metadata would be redundant with the micro/macro structure but that is also true for arrow.

Reply:
Should there be a general mechanism for providing a serializer to turn your object into bytes? For reading we allow you to request data in many different formats. Maybe the same should be true for writing?

Reply:
As a wire format, I think the server should accept just the array buffer because that is a more cross-language concept, where the `npy` format (with some dtype header in there) is a Python-specific thing.

I think it would be sensible for the server to _persist_ the data as npy though.